src/lib_json/json_value.cpp

```diff
@@ -614,6 +638,16 @@ bool Value::getString(char const** str, char const** cend) const {
   return true;
 }
 
+#if JSON_USE_SECURE_MEMORY
+inline std::string toStdString(Json::String in) {
+  return std::string(in.data(), in.data() + in.length());
+}
+#else
+inline std::string toStdString(std::string in) {
+  return in;
+}
+#endif
+
 std::string Value::asString() const {
```

Comment:
If we're changing the API to accept Json::String instead of std::string, then we might as well return Json::String too. In fact, that should never be a problem (except the loss of efficiency, since std::string is reference-counted on some platforms). The real problem is when a method is change to *accept* Json::String by const reference. Users will have to alter their code. I'm fine with that (since this is a compile-time choice) but it's something to note.

But for that reason, the default should be off, I think.